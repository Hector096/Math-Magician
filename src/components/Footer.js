import React from 'react';
import { Container } from 'react-bootstrap';
import '../App.css';

export default function Footer() {
  return (
    <div>
      <Container fluid className="bg-dark text-light p-3 footer text-center">
        &copy;
        {new Date().getFullYear()}
        - Math Magician by Hector - All Rights Reserved
      </Container>
    </div>
  );
}
