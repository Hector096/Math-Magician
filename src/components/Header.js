import React from 'react';
import { Navbar, Container, Nav } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';

export default function Header() {
  return (
    <Navbar bg="dark" variant="dark">
      <Container className="p-3">
        <Navbar.Brand href="/" className="h1">Math Magician</Navbar.Brand>
        <Nav className="ms-auto">
          <Link to="/" className="h5 me-4 text-decoration-none text-light">Home</Link>
          <Link to="/calculator" className="h5 me-4 text-decoration-none text-secondary">Calculator</Link>
          <Link to="/quote" className="h5 me-4 text-decoration-none text-secondary">Quote</Link>
        </Nav>
      </Container>
    </Navbar>
  );
}
