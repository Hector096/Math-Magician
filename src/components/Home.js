import React from 'react';

export default function Home() {
  return (
    <div>
      <h1 className="mt-5 text-center">Welcome to Our page!</h1>
      <p className="h4 p-5">
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Praesentium
        cum iste beatae eius repudiandae exercitationem in pariatur ratione
        delectus necessitatibus tempora nam culpa recusandae itaque odit quia
        consequuntur similique eveniet officiis, unde voluptatem autem quaerat?
        Molestiae magnam quos quis, quas, ex iure, pariatur fuga vel delectus
        autem voluptatibus laudantium temporibus unde possimus recusandae ut
        vitae. Nesciunt, natus cupiditate asperiores aspernatur reiciendis, iure
        repellat esse labore laudantium rerum quis accusamus quo voluptates
        beatae minima quasi corrupti voluptatibus! Similique delectus nihil
        nesciunt ducimus ea, quae nemo. Voluptas recusandae ipsam ipsum sapiente
        eius cumque nisi aliquid reiciendis harum iure sit soluta reprehenderit
        voluptatibus ducimus, provident.
        {' '}
        <br />
        repellendus repellat? Consequuntur quia
        praesentium eaque maiores nisi repellat in dolorem, ea deserunt
        exercitationem debitis officiis animi non nemo voluptatibus id officia
        reiciendis, perferendis facilis? Totam nisi quo, nam laudantium nesciunt
        consequuntur aliquid, modi dolore in consectetur reiciendis, beatae
        pariatur fugiat fuga assumenda. In dolores quisquam, commodi
        reprehenderit recusandae eius quia perspiciatis magni? Provident eaque
        suscipit nesciunt alias natus, at error a libero, nobis ipsa voluptate
      </p>
    </div>
  );
}
