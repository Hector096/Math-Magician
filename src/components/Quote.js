import React from 'react';
import '../App.css';

export default function Quote() {
  return (
    <div className="quote">
      <p className="h2 text-center">
        Mathematics is not about numbers, equations, computations,
        {' '}
        <br />
        or
        algorithms: it is about understanding. -William Paul Thruston
      </p>
    </div>
  );
}
