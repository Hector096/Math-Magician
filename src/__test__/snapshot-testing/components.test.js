import TestRenderer from 'react-test-renderer';
import Calculator from '../../components/calculator/calculator';
import Header from '../../components/Header';
import Home from '../../components/Home';
import Quote from '../../components/Quote';
import Footer from '../../components/Footer';

describe('test screenshots for all components', () => {
  test('matches Calculator snapshot', () => {
    const tree = TestRenderer.create(<Calculator />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Header snapshot', () => {
    const tree = TestRenderer.create(<Header />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Home snapshot', () => {
    const tree = TestRenderer.create(<Home />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Quote snapshot', () => {
    const tree = TestRenderer.create(<Quote />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches Footer snapshot', () => {
    const tree = TestRenderer.create(<Footer />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
