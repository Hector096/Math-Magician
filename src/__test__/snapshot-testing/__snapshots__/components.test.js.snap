// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test screenshots for all components matches Calculator snapshot 1`] = `
<div
  className="box"
>
  <div
    className="screen"
  >
     0
  </div>
  <div
    className="buttons"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="double"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`test screenshots for all components matches Footer snapshot 1`] = `
<div>
  <div
    className="bg-dark text-light p-3 footer text-center container-fluid"
  >
    ©
    2021
    - Math Magician by Hector - All Rights Reserved
  </div>
</div>
`;

exports[`test screenshots for all components matches Header snapshot 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <div
    className="p-3 container"
  >
    <a
      aria-current="page"
      className="h2 text-decoration-none text-light active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Math Magician
    </a>
    <div
      className="ms-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current="page"
        className="h5 me-4 link link-active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <a
        aria-current={null}
        className="h5 me-4 link"
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        aria-current={null}
        className="h5 me-4 link"
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </div>
  </div>
</nav>
`;

exports[`test screenshots for all components matches Home snapshot 1`] = `
<div>
  <h1
    className="mt-5 text-center"
  >
    Welcome to Our page!
  </h1>
  <p
    className="h4 p-5"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Praesentium cum iste beatae eius repudiandae exercitationem in pariatur ratione delectus necessitatibus tempora nam culpa recusandae itaque odit quia consequuntur similique eveniet officiis, unde voluptatem autem quaerat? Molestiae magnam quos quis, quas, ex iure, pariatur fuga vel delectus autem voluptatibus laudantium temporibus unde possimus recusandae ut vitae. Nesciunt, natus cupiditate asperiores aspernatur reiciendis, iure repellat esse labore laudantium rerum quis accusamus quo voluptates beatae minima quasi corrupti voluptatibus! Similique delectus nihil nesciunt ducimus ea, quae nemo. Voluptas recusandae ipsam ipsum sapiente eius cumque nisi aliquid reiciendis harum iure sit soluta reprehenderit voluptatibus ducimus, provident.
     
    <br />
    repellendus repellat? Consequuntur quia praesentium eaque maiores nisi repellat in dolorem, ea deserunt exercitationem debitis officiis animi non nemo voluptatibus id officia reiciendis, perferendis facilis? Totam nisi quo, nam laudantium nesciunt consequuntur aliquid, modi dolore in consectetur reiciendis, beatae pariatur fugiat fuga assumenda. In dolores quisquam, commodi reprehenderit recusandae eius quia perspiciatis magni? Provident eaque suscipit nesciunt alias natus, at error a libero, nobis ipsa voluptate
  </p>
</div>
`;

exports[`test screenshots for all components matches Quote snapshot 1`] = `
<div
  className="quote"
>
  <p
    className="h2 text-center"
  >
    Mathematics is not about numbers, equations, computations,
     
    <br />
    or algorithms: it is about understanding. -William Paul Thruston
  </p>
</div>
`;
